include/libvfio-user.h

```diff
@@ -241,6 +223,34 @@ typedef ssize_t (vfu_region_access_cb_t)(vfu_ctx_t *vfu_ctx, char *buf,
  * descriptors of these DM targets. This is something we can document and
  * demonstrate in a sample.
  *
+ * Areas that are accessed via such a mapping by definition do not invoke any
+ * given callback.  However, any access via VFIO_USER_REGION_READ/WRITE will do
+ * so, even if it's to a mappable area.
+ *
+ * A VFU_PCI_DEV_CFG_REGION_IDX region, corresponding to PCI config space, has
+ * special handling:
+ *
+ *  - the @size argument is ignored: the region size is always the size defined
+ *    by the relevant PCI specification
+ *  - all accesses to the standard PCI header (i.e. the first 64 bytes of the
+ *    region) are handled by the library
+ *  - all accesses to known PCI capabilities are handled by the library
+ *  - if no callback is provided, reads to other areas are a simple memcpy(),
+ *    and writes are an error
+ *  - otherwise, the callback is expected to handle the access
+ *
+ * @vfu_ctx: the libvfio-user context
+ * @region_idx: region index
+ * @size: size of the region
+ * @region_access: callback function to access region
+ * @flags: region flags (VFU_REGION_FLAG_)
```

Comment:
Yes, but I'm not sure what you mean in context - you'd like me to make changes so this is true?

Reply:
I think for each of these arguments we must say "ignored if the region is not mappable", rather than e.g. requesting that they have specific values (like we do for `nr_mmap_areas`).

Reply:
actually, I think we should be stricter: if !mappable, they must all be zero/-1. Will catch out silly mistakes easier.